src/backend/api/meals.js

```diff
@@ -1,12 +1,161 @@
 const express = require("express");
 const router = express.Router();
 const knex = require("../database");
+const { parse } = require("path");
+
+//NOTE: Everything works except for the availableReservations param route when combnining with other params.
+//SQL strict mode won't let groupping in this case - help needed :)
 
 router.get("/", async (request, response) => {
   try {
-    // knex syntax for selecting things. Look up the documentation for knex for further info
-    const titles = await knex("meals").select("title");
-    response.json(titles);
+    const maxPrice = request.query.maxPrice;
+    const availableReservations = request.query.availableReservations;
+    const title = request.query.title;
+    const createdAfter = request.query.createdAfter;
+    const limit = request.query.limit;
+    let results = knex("meals");
+
+    if (Object.keys(request.query).length == 0) {
+      const meals = await knex("meals").select("*");
+      response.json(meals);
+    } else {
+      let supportedParams = false;
+
+      if (maxPrice) {
+        if (!isNaN(parseInt(maxPrice))) {
+          supportedParams = true;
+          results = results.select("*").where("price", "<=", maxPrice);
+        } else {
+          supportedParams = false;
+          response
+            .status(400)
+            .send("Failed to parse max price. Please check the format.");
+          return;
+        }
+      }
+
+      if (availableReservations) {
+        if (availableReservations == "true") {
+          supportedParams = true;
+          results = results
+            .leftJoin("reservations", "meals.id", "reservations.meal_id")
+            .select(
+              "meals.id",
+              "meals.title",
+              "meals.description",
+              "meals.price",
+              "meals.max_reservations",
+              knex.raw(
+                "SUM(IFNULL(number_of_guests,0)) AS `guests_registered`"
+              ),
+              knex.raw(
+                "meals.max_reservations - SUM(IFNULL(number_of_guests,0)) AS `reservations_available`"
+              )
+            )
+            .where("max_reservations", ">", "number_of_guests")
+            .groupBy("meals.id")
+            .having(
+              knex.raw(
+                "(meals.max_reservations - SUM(IFNULL(number_of_guests,0)))>0"
+              )
```

Comment:
your solutions seems to be okay, I have seen many of your classmates had problem with this query. Try to discuss this in the class with the help of teacher/mentor.